### PR TITLE
test: add tests for RestTransportHandler

### LINKED ISSUE #278: [Test]: Run Express.js independent REST handler for Edge runtime
Currently [`rest_handler.spec.ts`](https://github.com/a2aproject/a2a-js/blob/main/test/server/rest_handler.spec.ts) depends on `express` so it had to be excluded from tests run on Edge runtime in #271. With this express agnostic [`rest_transport_handler.ts`](https://github.com/a2aproject/a2a-js/blob/main/src/server/transports/rest/rest_transport_handler.ts) is also not tested for Edge runtime.

It'd be useful to create a dedicated `rest_transport_handler.spec.ts` which can be run for Edge runtime.

### PR DESCRIPTION
**Pull request by tchapacan**
# Description

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/google-a2a/a2a-js/blob/main/CONTRIBUTING.md).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
  - Important Prefixes for [release-please](https://github.com/googleapis/release-please):
    - `fix:` which represents bug fixes, and correlates to a [SemVer](https://semver.org/) patch.
    - `feat:` represents a new feature, and correlates to a SemVer minor.
    - `feat!:`, or `fix!:`, `refactor!:`, etc., which represent a breaking change (indicated by the `!`) and will result in a SemVer major.
- [x] Ensure the tests and linter pass
- [ ] Appropriate docs were updated (if necessary)

Fixes #278 🦕

### CODE REVIEW

File: test/server/rest_transport_handler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+            parts: [
+              expect.objectContaining({
+                kind: 'file',
+                file: expect.objectContaining({ mimeType: 'application/pdf' }),
+              }),
+            ],
+          }),
+        }),
+        mockContext
+      );
```
Comment (ishymko) [inline test/server/rest_transport_handler.spec.ts]: The assertion looks the same as in the `snake_case` test case, let's merge both into one with payload being a parameter.

Same can be applied for other `snake_case`/`camelCase` tests here.

Comment (tchapacan) [reply test/server/rest_transport_handler.spec.ts]: Yep agreed it's better like this! Nice primitive to leverage with `it.each([])` didn't had the chance to use it a lot. It's maybe a little out of scope but I applied it as well in `rest_handler.spec.ts` as we have some snake-case and camelCase test that could be parametrized too. LMK if it's ok to treat it in this PR

Comment (ishymko) [reply test/server/rest_transport_handler.spec.ts]: > Nice primitive to leverage with `it.each([])`

Yeah indeed, haven't seen it before as well.

> I applied it as well in `rest_handler.spec.ts`

My thoughts exactly, thanks for this!
